tools/ipmonitor/README.md

```diff
@@ -0,0 +1,98 @@
+# Mozilla VPN - IPMonitor
+
+IPMonitor is a simple libpcap app to see which packages are sent and received
+outside the VPN tunnel. This is meant to be used as a debugging/testing tool
+and it's not included in the MozillaVPN app bundle.
+
+[![asciicast](https://asciinema.org/a/HZVQMHfQew3cm0lYUsLVZ78AT.png)](https://asciinema.org/a/HZVQMHfQew3cm0lYUsLVZ78AT)
+
+### How to compile it
+
+You must have the pcap library: https://www.tcpdump.org/
+
+On debian/ubuntu: `apt-get install libpcap0.8-dev`
+On macos: `brew reinstall libpcap`
+
+After this, run the following commands:
+
+1. [for macOn + brew only] `export PKG_CONFIG_PATH=/usr/local/opt/libpcap/lib/pkgconfig`
+2. `qmake && make`
+
+### How to fetch the list of servers from the MozillaVPN app
+
+First step is to retrieve the list of servers. You must have the mozillavpn
+client up and running. Then, run the following command:
+
+`mozillavpn servers -j > servers.json`
+
+Use `servers.json` as argument for the ipmonitor app.
+
+### How to run the app
+
+The app has an help menu:
+
+```
+$ ./ipmonitor -h 
+Usage: ./ipmonitor [options] server.list
+Mozilla VPN - IP Monitor
+
+Options:
+  -h, --help     Displays help on commandline options.
+  --help-all     Displays help including Qt specific options.
+  -v, --version  Displays version information.
+  -l <file>      Log non-VPN and non-LAN packages into this file
+
+Arguments:
+  server.list    The server list file
+```
+
+To run the app, pass the `server.json` as first argument. We use `sudo` to have
+the permissions to open the network interfaces.
+
+```
+$ sudo ./ipmonitor server.list
+make: Nothing to be done for `first'.
+Mozilla VPN - IP Monitor 0.1
+
+Processing the list of servers: 420 servers.
+Retrieving the interface list: done.
+The interfaces present on the system are:
+ 1 ap1 - 
+ 2 en0 - 
+ 3 awdl0 - 
+ 4 llw0 - 
+ 5 utun0 - 
+ 6 utun1 - 
+ 7 en5 - 
+ 8 en3 - 
+ 9 en4 - 
+ 10 en1 - 
+ 11 en2 - 
+ 12 bridge0 - 
+Which interface do you want to use?
+```
+
+At this point you have to choose the network interface you want to monitor.
+`en0` for MacOS and a `wlp*` interface on linux could be a good starting point.
+
+The app has this simple UI:
+```
+Total: 832 - VPN: 539 - LAN: 69 - OUT: 224
+D: < - Src:   192.168.0.128 - Dest:   193.32.249.69
```

Comment:
I would recommend breaking the `D:` line into multiple lines, because the addresses and direction can change too rapidly for a user to see when there is significant traffic. The lines I would recommend are (in whatever order seems most appropriate):
- Transmitted packets via VPN
- Received packets via VPN
- Transmitted leaks (packets transmitted on LAN interface to non-VPN destination address)
- Received leaks (packets received on LAN interface from non-VPN source address)
- Transmitted packets via LAN to local destinations (optional)
- Received packets via LAN from local sources (optional)

Such a format might look like:
```
Total: 832 - VPN: 539 - LAN: 69 - OUT: 224
VPN  Loc: Netherlands, Amsterdam
VPN  Tx: 192.168.0.128 --> 193.32.249.69
VPN  Rx: 193.32.249.69 --> 192.168.0.128
LEAK Tx: 192.168.0.128 --> 8.8.8.8
LEAK Rx: 8.8.8.8       --> 192.168.0.128
```

Reply:
If this is too complicated, then at least please have a separate line for transmit and receive.

Reply:
The issue here is that a single packet has a precise direction A -> B or B -> A. What you are suggesting here is a session.